frontend-v2/src/app/users/user-form.tsx

```diff
@@ -0,0 +1,347 @@
+'use client'
+
+import Link from 'next/link'
+import { useMemo } from 'react'
+import { useForm } from '@tanstack/react-form'
+import { z } from 'zod'
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+import { useRouter } from 'next/navigation'
+import toast from 'react-hot-toast'
+import { Checkbox } from '@/components/ui/checkbox'
+import { Input } from '@/components/ui/input'
+import { Label } from '@/components/ui/label'
+import { Button } from '@/components/ui/button'
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select'
+import { API_PATH, apiClient, Role, User, UserWithoutId } from '@/lib/api'
+import { Loader2 } from 'lucide-react'
+
+const roleOptions: Role[] = ['admin', 'organizer', 'attendance', 'non-sfbay']
```

Comment:
`Role` (exported from the api file) is already a `ZodEnum`. any reason to not just use that directly in the form schema here?

it can also be used below by doing `Role.options.map(...)`.

Reply:
done